plugins/admin_console/system_partner/lib/api/KalturaSystemPartnerConfiguration.php

```diff
@@ -678,9 +678,74 @@ private function copyMissingConversionProfiles(Partner $partner)
 		if($templatePartner)
 			myPartnerUtils::copyConversionProfiles($templatePartner, $partner, true);
 	}
-	
+
+	protected function isPermissionStatus($permissionName, $status)
+	{
+		foreach ($this->permissions as $permission)
+		{
+			if ($permission->name == $permissionName && $permission->status == $status)
+			{
+				return true;
+			}
+		}
+		return false;
+	}
+
+	protected function queryForCountingCategoriesWithTooManyPrivacyContexts($partnerId, $threshold)
+	{
+		$data =
+			[
+				"bool" =>
+				[
+					"filter" =>
+					[
+						[
+							"term" =>
+							[
+								"partner_status" =>
+								[
+									"value" => "p" . $partnerId . "s2"
+								]
+							]
+						],
+						[
+							"script" =>
+							[
+								"script" =>
+								[
+									"source" => "doc['privacy_contexts'].length > $threshold"
+								]
+							]
+						]
+					]
+				]
+			];
+		return $data;
+	}
+
 	public function validateForUpdate($sourceObject, $propertiesToSkip = array())
 	{
+		if ($this->isPermissionStatus(PermissionName::FEATURE_DISABLE_CATEGORY_LIMIT, PermissionStatus::ACTIVE))
+		{
+			$indexName = kBaseESearch::getElasticIndexNamePerPartner( ElasticIndexMap::ELASTIC_CATEGORY_INDEX, kCurrentContext::getCurrentPartnerId());
+			$params = array(
+				'index' => $indexName,
+				'type' => ElasticIndexMap::ELASTIC_CATEGORY_TYPE
+			);
+			$body = array();
+			$body['_source'] = false;
+			$body['query'] = $this->queryForCountingCategoriesWithTooManyPrivacyContexts($this->id, 2);
```

Comment:
i would change the name to something like buildExcedingPrivecyContextForCategoryQuery or something like that